host/Random.cpp

```diff
@@ -9,32 +9,27 @@ RandomNumberGenerator::~RandomNumberGenerator() {
 }
 
 PackedFloat RandomNumberGenerator::Generate() {
-    mpfr_t mpfr_num = {GenerateMpfr()};
+    mpfr_t mpfr_num;
+    GenerateMpfr(mpfr_num);
     PackedFloat num(mpfr_num);
     mpfr_clear(mpfr_num);
     return num;
 }
 
-__mpf_struct RandomNumberGenerator::GenerateGmp() {
-    mpf_t num;
+void RandomNumberGenerator::GenerateGmp(mpf_ptr num) {
     mpf_init2(num, kMantissaBits);
-    Generate(num);
-    return num[0];
 }
 
-__mpfr_struct RandomNumberGenerator::GenerateMpfr() {
-    mpfr_t num;
+void RandomNumberGenerator::GenerateMpfr(mpfr_ptr num) {
     mpfr_init2(num, kMantissaBits);
-    Generate(num);
-    return num[0];
```

Comment:
Isn't it missing the part where it actually generates anything? Right now it's doing nothing except initialization